### DISCUSSION TITLE
Pixi js. Displaying multiple animations at once.

### DISCUSSION THREAD

**Discussion by Kxarog** [pixijs/pixijs#9155] [Q&A]
pixi.js - v6.5.7
The problem is the following. I want to display several different animations on the screen, but I only have one displayed. Individually, each animation works as needed. Here is my code.

````
PIXI.settings.SCALE_MODE = PIXI.SCALE_MODES.NEAREST;
let loader = new PIXI.Loader();
const scene_render_1 = new PIXI.Container();
const scene_render_2 = new PIXI.Container();
const scene_render_3 = new PIXI.Container();
const scene_render_4 = new PIXI.Container();
const scene_render_5 = new PIXI.Container();
const scene_render_6 = new PIXI.Container();
const scene_render_7 = new PIXI.Container();
const scene_render_8 = new PIXI.Container();
const scene_render_9 = new PIXI.Container();
const scene_render_10 = new PIXI.Container();
const scene_render_11 = new PIXI.Container();
const scene_render_12 = new PIXI.Container();
let app = new PIXI.Application({
    width: 1280,
    height: 768,
    antialias: true,    
});
app.stage.addChild(
    scene_render_1, scene_render_2, scene_render_3,
    scene_render_4, scene_render_5, scene_render_6,
    scene_render_7, scene_render_8, scene_render_9,
    scene_render_10, scene_render_11, scene_render_12
);


scene_render_1.position.set(0, 0);
scene_render_2.position.set(0, 0);
scene_render_3.position.set(0, 0);
scene_render_4.position.set(0, 0);
scene_render_5.position.set(0, 0);
scene_render_6.position.set(0, 0);
scene_render_7.position.set(0, 0);
scene_render_8.position.set(0, 0);
scene_render_9.position.set(0, 0);
scene_render_10.position.set(0, 0);
scene_render_11.position.set(0, 0);
scene_render_12.position.set(0, 0);
document.body.appendChild( app.view );
PIXI.Loader.shared.add("images/animation_plita.json").load(animation_plita);
function animation_plita() {
  let sheet_animation_plita = PIXI.Loader.shared.resources["images/animation_plita.json"].spritesheet;
  avatar_animation_plita = new PIXI.AnimatedSprite(sheet_animation_plita.animations["idle_plita"]);
  avatar_animation_plita.animationSpeed = 0.15
  avatar_animation_plita.scale.set(2);
  avatar_animation_plita.position.set(180, 576);
  avatar_animation_plita.play();
  scene_render_3.addChild(avatar_animation_plita);
}
PIXI.Loader.shared.add("images/animation_altar.json").load(animation_altar);
function animation_altar() {
  let sheet_animation_altar = PIXI.Loader.shared.resources["images/animation_altar.json"].spritesheet;
  avatar_animation_altar = new PIXI.AnimatedSprite(sheet_animation_altar.animations["idle_altar"]);
  avatar_animation_altar.animationSpeed = 0.15
  avatar_animation_altar.scale.set(2);
  avatar_animation_altar.position.set(400, 200);
  avatar_animation_altar.play();
  scene_render_3.addChild(avatar_animation_altar);
}
````

**Comment by SuperSodaSea** [comment]
It seems that you are only using `scene_render_3`?

**Reply by Kxarog** [reply]
No. I deleted the extra code that is not relevant to the question for a better understanding. Splitting into scenes is necessary for correct rendering of the background / images / text/ (it's more convenient for me).

**Reply by domis86** [reply]
@Kxarog  i think @SuperSodaSea  meant this:

```
scene_render_3.addChild(avatar_animation_plita);
...
scene_render_3.addChild(avatar_animation_altar);
```

should be maybe like this? :

```
scene_render_3.addChild(avatar_animation_plita);
...
scene_render_4.addChild(avatar_animation_altar);
```

**Reply by SuperSodaSea** [reply]
I re-read the code and I think it should be OK to add multiple sprites to a single container. Maybe there is something else going wrong.

**Reply by Kxarog** [reply]
It's not about the containers. It's just the order of rendering, no more. I clearly made a video on the problem that would be more understandable.
[https://youtu.be/M61lxa4xKyY](url)

And in this old video you can see that all the other elements (pictures / backgrounds / fonts, etc.) are displayed without problems. And therefore it's not about the scenes
[https://youtu.be/RQWXAlZkR4o](url)

**Reply by domis86** [reply]
can you prepare simple reproduction of this problem in https://www.pixiplayground.com , jsfiddle, codesandbox or similar?

**Reply by Kxarog** [reply]
I couldn't solve the problem, but I found some kind of analog (not perfect, but it will work for me).
pixi.js - v7.1.2
```

function open_1_1() {
    PIXI.Assets.load('images/animation_plita.json').then(() => {
        let explosionTextures = [];
        let i;
        for (i = 1; i < 21; i++) {
            explosionTextures.push(PIXI.Texture.from(`idle_${i}.png`));
        }

        let explosion = new PIXI.AnimatedSprite(explosionTextures);
        explosion.animationSpeed = 0.14;
        explosion.position.set(368, 380);
        explosion.scale.set(2);
        explosion.play();
        scene_render_12.addChild(explosion);
    });    
}
    
open_1_1();  

```
I changed the animation code and now several animations are displayed on the screen at once. But unfortunately there is a problem there too (not critical for me, but it is there) which I clearly showed in the video.

https://youtu.be/es07C-bYkVE

I am not an expert in programming and pixi, but it seems to me that there is some kind of problem in the loader, since both versions of the code were taken from official documentation and examples.(that is, the chance of an error on my part is minimal)

The first option from https://api.pixijs.io/@pixi/sprite-animated/PIXI/AnimatedSprite.html

The second of https://pixijs.io/examples/#/sprite/animatedsprite-explosion.js
__________________________________________________________________________________
New information. The bug shown in the video was in this line and in the file .json

The name of the frames in the animation should be different.
For example , if one file .json each frame should be named idle_1, idle_2, idle_3, etc., then in another file idle__1, idle__2, idle__3, etc. In the third idle_plita_1, idle_plita_3, idle_plita_3, etc. In the code itself, this line also needs to be changed depending on the name that you have registered.

`explosiontexture.push(PIXI.Texture.from(`idle_${i}.png`));`

https://youtu.be/vEaoc1SvodA
